contracts/governance/Vesting/VestingLogic.sol

```diff
@@ -158,6 +161,7 @@ contract VestingLogic is IVesting, VestingStorage, ApprovalReceiver {
         /// @dev Don't change FOUR_WEEKS to TWO_WEEKS, a lot of vestings already deployed with FOUR_WEEKS
         ///		workaround found, but it doesn't work with TWO_WEEKS
         for (uint256 i = startDate + cliff; i <= end; i += FOUR_WEEKS) {
+            if (counter >= maxVestingWithdrawIterations) break;
```

Comment:
1) below  
stake = staking.getPriorUserStakeByDate(address(this), i, block.number - 1);
will double gas consuming when running the function again after maxVestingWithdrawIterations is reached
pls consider using (or adding if there is none) function to return the earliest stake available not to cycle through already covered (withdrawn) dates 
2) we need to let the user know that the wthdrawal is incomplete - pls consider adding an event